components/locale/src/fallback/algorithms.rs

```diff
@@ -219,30 +296,60 @@ mod tests {
             input: "en-u-sd-usca",
             requires_data: true,
             expected_language_chain: &["en-u-sd-usca", "en"],
+            expected_script_chain: &[
+                "en-u-sd-usca",
+                "en",
+                "und-Latn-u-sd-usca",
+                "und-Latn",
+                "und-u-sd-usca",
+            ],
             expected_region_chain: &["en-US-u-sd-usca", "en-US", "und-US-u-sd-usca", "und-US"],
         },
         TestCase {
             input: "en-Latn-u-sd-usca",
             requires_data: true,
             expected_language_chain: &["en-u-sd-usca", "en"],
+            expected_script_chain: &[
+                "en-Latn-u-sd-usca",
+                "en-Latn",
+                "en",
+                "und-Latn-u-sd-usca",
+                "und-Latn",
+                "und-u-sd-usca",
+            ],
```

Comment:
Issue: whether the locale starts as `en-Latn-u-sd-usca` or `en-u-sd-usca` should not affect the reachable locales, because those two locales are identical. They should both go through `en-Latn-u-sd-usca` or `en-u-sd-usca`. Or maybe both? (Probably only for a non-default script)

Also, I'm not sure `und-u-sd-usca` should be reachable. It isn't reachable in the region fallback chain.

I lean toward:

1. `en-US-u-sd-usca`
2. `en-US`
3. `en`
7. `und-Latn-US-u-sd-usca`
8. `und-Latn-US`
9. `und-Latn`
10. `und`

Reply:
This chain containing `-US` doesn't seem to be for the locale `en-Latn-u-sd-usca`?

Reply:
`en-Latn-u-sd-usca` and `en-u-sd-usca` have the same chain, so I'm resolving this.